Xamarin.Forms.Core/Command.cs

```diff
@@ -8,31 +9,51 @@ public sealed class Command<T> : Command
 		public Command(Action<T> execute) 
 			: base(o =>
 			{
-				if (o is T)
+				if (IsValidParameter(o))
 				{
 					execute((T)o);
 				}
 			})
 		{
 			if (execute == null)
 			{
-				throw new ArgumentNullException("execute");
+				throw new ArgumentNullException(nameof(execute));
 			}
 		}
 
 		public Command(Action<T> execute, Func<T, bool> canExecute) 
 			: base(o =>
 			{
-				if (o is T)
+				if (IsValidParameter(o))
 				{
 					execute((T)o);
 				}
-			}, o => o is T && canExecute((T)o))
+			}, o => IsValidParameter(o) && canExecute((T)o))
 		{
 			if (execute == null)
-				throw new ArgumentNullException("execute");
+				throw new ArgumentNullException(nameof(execute));
 			if (canExecute == null)
-				throw new ArgumentNullException("canExecute");
+				throw new ArgumentNullException(nameof(canExecute));
+		}
+
+		static bool IsValidParameter(object o)
+		{
+			if (o != null)
+			{
+				// The parameter isn't null, so we don't have to worry whether null is a valid option
+				return o is T;
+			}
+
+			var t = typeof(T);
+
+			// The parameter is null. Is T Nullable?
+			if (Nullable.GetUnderlyingType(t) != null)
+			{
+				return true;
+			}
+
+			// Not a Nullable, if it's a value type then null is not valid
+			return !t.GetTypeInfo().IsValueType;
```

Comment:
I'd swap the 2 checks. first check if it's a reference type, and if it's not, start to wonder about Nullable...